### DISCUSSION TITLE
LTSP has gone SLOW!

### DISCUSSION THREAD

**Discussion by danboid** [ltsp/ltsp#840] [Q&A] [answered]
Our Ubuntu 20.04 based (both the server and the image were 20.04) LTSP server was running fine for over a year then about 1-2 months ago it slowed down massively, to the point where even if only one LTSP client is logged it it takes over a minute for it to load a small app like xfce4-terminal.

I was hoping that whatever the problem is might get fixed by updating our LTSP server to Ubuntu 22.04 but I've done that today and LTSP is still running veeery slow on the clients. The upgrade seemed to go fine otherwise and it does seem a little faster than under 20.04 but its still painfully slow.

I plugged a USB HD into one of the LTSP clients and I downloaded an ISO via wget from the LTSP server. I was still getting around 110 MB/s transfer rate from the LTSP, like I was before it slowed down.

htop doesn't indicate any processes are heavily hogging the CPU of the LTSP server so I'm not sure why its running so slowly now after running fine for a year or so? Maybe there's an NFS setting I can try adjusting that might help?

**Comment by danboid** [comment]
I'm pretty sure this has been caused by sanoid failing to auto prune snapshots so I'm currently destroying the 650K snapshots that amassed on the pool and then I'll be swapping sanoid for zfs-auto-snapshot, which I know prunes snapshots properly. We have another server with a similar number of users but that only has about 50K snapshots across the pool because its using z-a-s.

**Comment by danboid** [comment]
It took weeks but I've finally destroyed all ZFS snapshots on all datasets on our LTSP server but its still running very slowly so I don't think ZFS is the issue here. I've disabled snapshots until I've worked out what the problem is. This server was running LTSP fine using ZFS for about a year before it slowed to a crawl a few months ago. I don't want to to have re-install it and we could end up with the same issue.

The LTSP clients still boot in a reasonable time but after logging in, everything is very slow. It takes about a minute or so to load the XFCE desktop then to load Chrome takes slightly over three minutes on a modern i7 desktop with 32 GB RAM with no other programs running.

How might I go about troubleshooting this? I tried using wget to download an ISO from the LTSP server to a USB drive attached to client and that worked fine but that isn't using NFS / sshfs like LTSP.

Thanks

**Comment by danboid** [comment]
As alkisg advised, I ran `hdparm -t` on all of the SSDs in my pool and one of them was running notably slower than the others so I'm running a long smartctl test on that now but it looks like I'll need to at least swap that disk out to get the performance back.

**Comment by danboid** [comment, marked as answer]
It seems that a better way (better than hdparm) to test the health of SATA SSDs under Linux is to use one of these smartctl TBW (Total Bytes Written) scripts:

https://askubuntu.com/questions/865792/how-can-i-monitor-the-tbw-on-my-samsung-ssd

If the TBW is less than 90% you probably need to replace the disk.

I have requested that Zabbix add TBW support to their S.M.A.R.T monitoring template:

https://support.zabbix.com/browse/ZBXNEXT-8700

**Reply by danboid** [reply]
Actually, there's no need for that TBW script.

For Samsung SATA SSDs you can get the wear levelling count:

smartctl -a /dev/sda | grep Wear | awk '{print $4}'

For NVMe:

smartctl -a /dev/nvme0n1 | grep Percentage | awk '{print $3}'

**Comment by brcisna** [comment]
@danboid 
Did you ever get this slow down of your LTSP network, actually figured out for certain. So many times these threads go dead, without anyone actually saying what the fix was.

**Reply by danboid** [reply]
I did fix the issue yes. We store our LTSP home dirs on a RAIDZ2 ZFS pool of 6 SSDs. At least 2 of the SSDs in this pool had a wear level of over 10% according to smartctl. I replaced those SSDs and the LTSP server regained full speed.
